syntax = "proto3";
package gateway_service;
option cc_generic_services = true;


message LoginRequest
{
    string account = 1;
    string password = 2;
}

message LoginResponse 
{
    enum Status
    {
        SUCCESS = 0;
        UNKNOWN_ERROR = 1;
        UNKNOWN_ACCOUNT = 2;
        INVALID_PASSWORD = 3;
        ACCOUNT_LOCKED = 4;
        ACCOUNT_DISABLED = 5;
    }
    Status status = 1;
    string token = 2;
}

message LogoutRequest
{
    string token = 1;
    string account = 2;
}

message LogoutResponse
{
    enum Status
    {
        SUCCESS = 0;
        UNKNOWN_ERROR = 1;
        INVALID_TOKEN = 2;
        TOKEN_EXPIRED = 3;
    }
    Status status = 1;
}

service LoginService
{
    rpc Login(LoginRequest) returns (LoginResponse);
    rpc Logout(LogoutRequest) returns (LogoutResponse);
}